src/System.IO.FileSystem/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -13,11 +13,6 @@ public sealed partial class SafeFileHandle : SafeHandle
     {
         public SafeFileHandle(IntPtr preexistingHandle, bool ownsHandle) : this(ownsHandle)
         {
-            long longHandle = (long)preexistingHandle;
```

Comment:
I effectively moved the checks to IsInvalid rather than doing it at construction time.  As far as I can tell, that's more the prescribed pattern / how we do things on Windows, and doing it the original way was causing differences in behavior between Windows and Unix.